src/task.rs

```diff
@@ -3,7 +3,50 @@
 use std::pin::Pin;
 
 use futures::prelude::*;
-use futures::task::{Context, Poll};
+use futures::future::FutureObj;
+use futures::task::{Context, Poll, Spawn, SpawnError};
+
+/// A [`Spawn`] handle to runtime's thread pool for spawning futures.
+///
+/// This allows integrating runtime with libraries based on explicitly passed spawners.
+#[derive(Debug)]
+pub struct Spawner {
```

Comment:
An alternate name for this could possibly be `Executor` or `ExecutorHandle`. But I'm not at all clear on what the best name is, and `Spawner` seems decent so happy to go with either name!